src/input_common/tas/tas_input.h

```diff
@@ -0,0 +1,163 @@
+            returns += ", Y";
+        if ((button & static_cast<u32>(TasInput::TasButton::STICK_L)) != 0)
+            returns += ", STICK_L";
+        if ((button & static_cast<u32>(TasInput::TasButton::STICK_R)) != 0)
+            returns += ", STICK_R";
+        if ((button & static_cast<u32>(TasInput::TasButton::TRIGGER_L)) != 0)
+            returns += ", TRIGGER_L";
+        if ((button & static_cast<u32>(TasInput::TasButton::TRIGGER_R)) != 0)
+            returns += ", TRIGGER_R";
+        if ((button & static_cast<u32>(TasInput::TasButton::TRIGGER_ZL)) != 0)
+            returns += ", TRIGGER_ZL";
+        if ((button & static_cast<u32>(TasInput::TasButton::TRIGGER_ZR)) != 0)
+            returns += ", TRIGGER_ZR";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_PLUS)) != 0)
+            returns += ", PLUS";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_MINUS)) != 0)
+            returns += ", MINUS";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_LEFT)) != 0)
+            returns += ", LEFT";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_UP)) != 0)
+            returns += ", UP";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_RIGHT)) != 0)
+            returns += ", RIGHT";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_DOWN)) != 0)
+            returns += ", DOWN";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_SL)) != 0)
+            returns += ", SL";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_SR)) != 0)
+            returns += ", SR";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_HOME)) != 0)
+            returns += ", HOME";
+        if ((button & static_cast<u32>(TasInput::TasButton::BUTTON_CAPTURE)) != 0)
+            returns += ", CAPTURE";
+        return returns.length() != 0 ? returns.substr(2) : "";
+    }
+
+    void RefreshTasFile();
+    void LoadTasFiles();
+    void RecordInput(u32 buttons, std::array<std::pair<float, float>, 2> axes);
+    void UpdateThread();
+    std::string GetStatusDescription();
+
+    InputCommon::ButtonMapping GetButtonMappingForDevice(const Common::ParamPackage& params) const;
+    InputCommon::AnalogMapping GetAnalogMappingForDevice(const Common::ParamPackage& params) const;
+    [[nodiscard]] const TasData& GetTasState(std::size_t pad) const;
+
+private:
+    struct TASCommand {
+        u32 buttons{};
+        TasAnalog l_axis{};
+        TasAnalog r_axis{};
+    };
+    void LoadTasFile(int playerIndex);
+    void WriteTasFile();
+    TasAnalog ReadCommandAxis(const std::string line) const;
+    u32 ReadCommandButtons(const std::string line) const;
+    std::string WriteCommandButtons(u32 data) const;
+    std::string WriteCommandAxis(TasAnalog data) const;
+    std::pair<float, float> flipY(std::pair<float, float> old) const;
```

Comment:
```suggestion
    std::pair<float, float> FlipY(std::pair<float, float> old) const;
```

This can likely also be a static member function inside of the cpp file.

Reply:
Does that mean completely removing the function prototype from the header file and just making it a static method in the `cpp` like the `Debug[...]`-methods?

Reply:
Yeah